src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AmendReferralService.kt

```diff
@@ -1,44 +1,59 @@
 package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
 
+import org.springframework.data.repository.findByIdOrNull
 import org.springframework.http.HttpStatus
 import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken
 import org.springframework.stereotype.Service
 import org.springframework.web.server.ResponseStatusException
+import org.springframework.web.server.ServerWebInputException
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.authorization.UserMapper
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.AmendComplexityLevelDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.AmendDesiredOutcomesDTO
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.ReferralAmendmentDetails
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEventPublisher
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Changelog
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.SelectedDesiredOutcomesMapping
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ChangelogRepository
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ServiceCategoryRepository
 import java.time.OffsetDateTime
 import java.util.UUID
 import javax.transaction.Transactional
 
+enum class AmendTopic {
+  COMPLEXITY_LEVEL,
+  DESIRED_OUTCOMES
+}
+
 @Service
 @Transactional
 class AmendReferralService(
   private val referralEventPublisher: ReferralEventPublisher,
   private val changelogRepository: ChangelogRepository,
   private val referralRepository: ReferralRepository,
+  private val serviceCategoryRepository: ServiceCategoryRepository,
   private val userMapper: UserMapper,
   private val referralService: ReferralService
 ) {
 
-  fun updateComplexityLevel(referral: Referral, update: AmendComplexityLevelDTO, authentication: JwtAuthenticationToken): AmendComplexityLevelDTO? {
+  fun updateComplexityLevel(referral: Referral, update: AmendComplexityLevelDTO, serviceCategoryId: UUID, authentication: JwtAuthenticationToken): Changelog? {
+
+    if (referral.approvedActionPlan != null) {
+      throw ServerWebInputException("desired outcomes cannot be updated: the action plan is already approved")
```

Comment:
should say 'complexity level cannot be updated'

Reply:
Its a copy-paste error, will update it